controllers/stock_picking_batch.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+
+from odoo import http
+from odoo.http import request
+
+from .main import UdesApi
+
+
+def _get_single_batch_info(batch):
+    if not batch:
+        return {}
+
+    info = batch.get_info()
+    assert len(info) == 1, "expected exactly one batch"
```

Comment:
I would assert len(batch) before calling get_info(), this way we avoid to call get_info() if we have more than one batch.

Reply:
I'm not sure I understand the point here and I may be missing something :P

Assertions are meant for logic errors, so I don't see a benefit in avoiding the `get_info` call.
By having the assertion on `info`, you're asserting the function assumptions on both `batch` (the recordset should be a singleton) and what `get_info` actually does.
So I don't think we should move it, nor add another assertion.